src/parser/ast/Node.h

```diff
@@ -90,24 +90,25 @@ enum ASTNodeType {
     BinaryExpressionBitwiseOr,
     BinaryExpressionBitwiseXor,
     BinaryExpressionDivison,
+    /* Note: These 8 types must be in this order */
     BinaryExpressionEqual,
+    BinaryExpressionNotEqual,
+    BinaryExpressionStrictEqual,
+    BinaryExpressionNotStrictEqual,
     BinaryExpressionGreaterThan,
     BinaryExpressionGreaterThanOrEqual,
+    BinaryExpressionLessThan,
+    BinaryExpressionLessThanOrEqual,
```

Comment:
Could you add static_assert about testing this(These 8 types must be in this order)?
Otherthings are look great to me